kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionImpl.java

```diff
@@ -386,11 +383,25 @@ private long doCommit(
     resolvedDomainMetadatas.forEach(
         dm -> metadataActions.add(createDomainMetadataSingleAction(dm.toRow())));
 
-    try (CloseableIterator<Row> stageDataIter = dataActions.iterator()) {
+    try (CloseableIterator<Row> userStageDataIter = dataActions.iterator()) {
+
+      CloseableIterator<Row> completeFileActionIter;
```

Comment:
final? This makes it clearer that we are trying to assign it in either block of the next `if` statement -- and we don't want to re-assign it later